packages/core-web/src/rpc.ts

```diff
@@ -0,0 +1,207 @@
+import type {
+  CancelFunction,
+  JSONValue,
+  ModuleKind,
+  RpcRequest,
+  RpcResponse,
+  RpcRequestFull,
+  RpcResponseFull,
+} from './types'
+import { logger } from './utils/logger'
+
+export interface RpcTransport {
+  sendRequest(request: RpcRequestFull): void
+  destroy(): void
+}
+
+export interface RpcTransportInit {
+  init(
+    onRpcResponse: (response: RpcResponseFull) => void,
+  ): Promise<RpcTransport>
+}
+
+// Handles communication with the wasm worker
+// TODO: Move rpc stream management to a separate "SubscriptionManager" class
+export class RpcClient {
+  private transport?: RpcTransport
+  private transportInit: RpcTransportInit
+  private requestCounter = 0
+  private requestCallbacks = new Map<number, (response: RpcResponse) => void>()
+  private initPromise?: Promise<void>
+  private clientName: string | undefined
```

Comment:
try to remove client name from here and pass it as argument to method where this is needed

Reply:
looks like only rpcStream and rpcSingle need it.